### PR TITLE
New user admin routes and request classes

### PR DESCRIPTION
**Pull request by denniscunningham**
PEPPER-1007

Reworked user admin routes and user admin service to support a different RESTful interface, largely to provide endpoints that are easier to the UI to work with. See [DSM User Administration Routes](https://docs.google.com/document/d/1RK8qeOY48Rt5cslgdhO5LrtN2C88im3uRugzj_3KkcI/edit).

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UserAdminService.java [hunk 1]
```diff
@@ -128,26 +128,98 @@ public StudyRoleResponse getStudyRoles() {
...
+
+        List<String> addRoles = req.getAddRoles();
+        List<String> removeRoles = req.getRemoveRoles();
+        boolean hasAddRoles = CollectionUtils.isNotEmpty(addRoles);
+        boolean hasRemoveRoles = CollectionUtils.isNotEmpty(removeRoles);
+
+        if (hasAddRoles) {
+            // ensure no union of add and remove roles
+            if (hasRemoveRoles && CollectionUtils.containsAny(addRoles, removeRoles)) {
+                throw new DsmInternalError("Invalid user request: Cannot add and remove the same user");
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UserAdminService.java]: Wondering if message can be more clear like: "Cannot add and remove same user  to the same role" ?  OR "Cannot add and remove same role to the same user" ?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UpdateUserRoleRequest.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Data;
+
+@Data
+public class UpdateUserRoleRequest {
+
+    private final List<String> users;
+    private final List<String> addRoles;
+    private final List<String> removeRoles;
+
+    public UpdateUserRoleRequest(List<String> users, List<String> addRoles, List<String> removeRoles) {
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UpdateUserRoleRequest.java:14]: Is this class to add and remove all the roles to ALL the users ?
I mean ALL users have same exact changes (add and remove passed roles), right ?

**Review by ssettipalli — APPROVED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UpdateUserRoleRequest.java (hunk 2 above)
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UpdateUserRoleRequest.java:14]: That is correct. It is unlikely the UI will use this, at least in the first iteration, but this is useful for bulk operations so I kept it around. (I actually used the former version of the associated endpoint for many of the permission updates over the past couple sprints.)

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UserAdminService.java (hunk 1 above)
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UserAdminService.java]: Ah, good catch... this is not a good message. Will fix.

**Review by denniscunningham — COMMENTED** [review]
@ssettipalli Thank you for reviewing!
I will make the change you suggested.

**Pull request merged**